src/main/java/de/unijena/cheminf/mortar/model/fragmentation/algorithm/AlkylStructureFragmenter.java

```diff
@@ -0,0 +1,767 @@
+        this.settingNameTooltipTextMap.put(this.dissectRingsSetting.getName(),
+                Message.get("AlkylStructureFragmenter.dissectRingsSetting.tooltip"));
+        this.preserveRingMaxSizeSetting = new SimpleEnumConstantNameProperty(this, "Preserve ring max size setting (preserves if Dissect rings setting is set false)",
+                AlkylStructureFragmenter.Preserve_Ring_MAX_SIZE_OPTION_DEFAULT.name(),
+                PreserveRingMaxSizeOption.class) {
+            @Override
+            public void set(String newValue) throws NullPointerException, IllegalArgumentException {
+                try {
+                    super.set(newValue);
+                } catch (NullPointerException | IllegalArgumentException anException) {
+                    AlkylStructureFragmenter.this.logger.log(Level.WARNING, anException.toString(), anException);
+                    GuiUtil.guiExceptionAlert("Illegal Argument", "Illegal Argument was set", anException.toString(), anException);
+                    //re-throws the exception to properly reset the binding
+                    throw anException;
+                }
+            }
+        };
+        this.settingNameTooltipTextMap.put(this.preserveRingMaxSizeSetting.getName(),
+                Message.get("AlkylStructureFragmenter.preserveRingMaxSizeSetting.tooltip"));
+        this.preserveRingSystemSetting = new SimpleBooleanProperty(this, "Preserve ring system setting (preserves if Dissect rings setting is set false)",
+                AlkylStructureFragmenter.PRESERVE_RING_SYSTEM_OPTION_DEFAULT);
+        this.settingNameTooltipTextMap.put(this.preserveRingSystemSetting.getName(),
+                Message.get("AlkylStructureFragmenter.preserveRingSystemSetting.tooltip"));
+        this.preserveRingSystemMaxSetting = new SimpleIntegerProperty(this, "Preserve ring system up to maximum size",
+                AlkylStructureFragmenter.PRESERVE_RING_SYSTEM_MAX_OPTION_DEFAULT);
+        this.settingNameTooltipTextMap.put(this.preserveRingSystemMaxSetting.getName(),
+                Message.get("AlkylStructureFragmenter.preserveRingSystemMaxSetting.tooltip"));
+        //
+        this.settings = new ArrayList<Property>(7);
+        this.settings.add(this.fragmentSaturationSetting);
+        this.settings.add(this.chainFragmentLengthSetting);
+        this.settings.add(this.dissectSpiroCarbonSetting);
+        this.settings.add(this.dissectRingsSetting);
+        this.settings.add(this.preserveRingMaxSizeSetting);
+        this.settings.add(this.preserveRingSystemSetting);
+        this.settings.add(this.preserveRingSystemMaxSetting);
+    }
+    //</editor-fold>
+    //
+    //<editor-fold desc="Get Settings">
+    /**
+     * Returns a list of all available settings represented by properties for the given fragmentation algorithm.
+     *
+     * @return list of settings represented by properties
+     */
+    @Override
+    public List<Property> settingsProperties() {
```

Comment:
The doc strings of overridden methods does not need to be copied, unless you want to add something specific.